device/core/src/device_client.ts

```diff
@@ -298,6 +366,6 @@ export class Client extends InternalClient {
 
     /*Codes_SRS_NODE_DEVICE_CLIENT_16_090: [The `fromAuthenticationProvider` method shall pass the `authenticationProvider` object passed as argument to the transport constructor.]*/
     /*Codes_SRS_NODE_DEVICE_CLIENT_16_091: [The `fromAuthenticationProvider` method shall return a `Client` object configured with a new instance of a transport created using the `transportCtor` argument.]*/
-    return new Client(new transportCtor(authenticationProvider), null, new BlobUploadClient(authenticationProvider));
+    return new Client(new transportCtor(authenticationProvider), null, new BlobUploadClient(authenticationProvider), new DefaultFileUploadApi(authenticationProvider));
```

Comment:
>new BlobUploadClient(authenticationProvider), new DefaultFileUploadApi(authenticationProvider) [](start = 71, length = 94)

if you instantiate the FileUploadApi class but don't pass it to the BlobUploadClient it's going to instantiate one on its own which might waste resources. I think you should pass the fileuploadApi you create to the blob upload client.